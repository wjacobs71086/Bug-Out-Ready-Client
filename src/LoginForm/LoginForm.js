import React, { Component } from "react";
import TokenService from "../Services/token-service";
import AuthApiService from "../Services/auth-api-service";

export default class LoginForm extends Component {
  static defaultProps = {
    onLoginSuccess: () => {}
  };

  state = {
    error: null,
    user_name: null,
    password: null,

    touched: false
  };

  handleSubmitJwtAuth = ev => {
    ev.preventDefault();
    this.setState({
      error: null
    });
    const { user_name, password } = ev.target;
    AuthApiService.postLogin({
      user_name: user_name.value,
      password: password.value
    })

      .then(res => {
        user_name.value = "";
        password.value = "";
        TokenService.saveAuthToken(res.authToken);
        this.props.onLoginSuccess();
      })
      .catch(res => {
        this.setState({
          error: res.error
        });
      });
  };

  render() {
    const { error } = this.state;
    return (
      <form className="LoginForm" onSubmit={this.handleSubmitJwtAuth}>
        <div role="alert">{error && <p className="red">{error}</p>}</div>

        <input
          required
          name="user_name"
          placeholder="Username"
          className="loginInput"
        ></input>

        <input
          required
          name="password"
          placeholder="Password"
          type="password"
          className="loginInput"
        ></input>
        <button type="submit" className="login">
          Login
        </button>
      </form>
    );
  }
}
