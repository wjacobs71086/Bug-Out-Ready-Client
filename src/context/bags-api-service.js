import config from '../config'
import TokenService from './token-service';

const BagsApiService = {
  getThings() {
    return fetch(`${config.API_ENDPOINT}/bags`, {
      headers: {
        'authorization': `bearer ${TokenService.getAuthToken()}`,
      },
    })
      .then(res =>
        (!res.ok)
          ? res.json().then(e => Promise.reject(e))
          : res.json()
      )
  },
  getThing(bagId) {
    console.log()
    return fetch(`${config.API_ENDPOINT}/bag-home/${bagId}`, {
      headers: {
        'authorization': `bearer ${TokenService.getAuthToken()}`,
      },
    })
      .then(res => {
        (!res.ok)
          ? res.json().then(e => Promise.reject(e))
          : res.json()
      })
  },
  getBagsItems(bagId) {
    console.log(`${config.API_ENDPOINT}/bag-home/${bagId}`);
    return fetch(`${config.API_ENDPOINT}/bag-home/${bagId}`, {
      headers: {
        'authorization': `bearer ${TokenService.getAuthToken()}`,
      },
    })
      .then(res =>
        (!res.ok)
          ? res.json().then(e => Promise.reject(e))
          : res.json()
      )
  }

}

export default BagsApiService;