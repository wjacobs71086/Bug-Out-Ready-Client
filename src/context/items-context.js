import React, { Component } from 'react';


const ItemsListContext = React.createContext({
  itemsList: [],
  bagsList: [],
  userId: [],
  error: null,
  setError: () => {},
  clearError: () => {},
  setItemsList: () => {},
  setBagsList: () => {},
})
export default ItemsListContext

export class ItemsListProvider extends Component {
  state = {
    itemsList: [],
    bagsList: [],
    error: null,
  };

  setItemsList = itemsList => {
    this.setState({ itemsList })
  }

  setBagsList = bagsList => {
    this.setState({ bagsList })
  }

  setError = error => {
    console.error(error)
    this.setState({ error })
  }

  clearError = () => {
    this.setState({ error: null })
  }

  render() {
    const value = {
      itemsList: this.state.itemsList,
      error: this.state.error,
      setError: this.setError,
      clearError: this.clearError,
      setItemsList: this.setItemsList,
      setBagsList: this.setBagsList,
      bagsList: this.state.bagsList,
      userId: this.state.userId,
    }
    return (
      <ItemsListContext.Provider value={value}>
        {this.props.children}
      </ItemsListContext.Provider>
    )
  }
}