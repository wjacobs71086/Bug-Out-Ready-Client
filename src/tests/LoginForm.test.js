import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import LoginForm from '../LoginForm/LoginForm';
import { BrowserRouter } from 'react-router-dom'
import { ItemsListProvider } from '../context/items-context';
import { Route
} from 'react-router-dom'

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <BrowserRouter>
            <ItemsListProvider>
                <Route
                    component={LoginForm}
                />
            </ItemsListProvider>
        </BrowserRouter>
        , div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
      .create(        
      <BrowserRouter>
        <ItemsListProvider>
            <Route
                component={LoginForm}
            />
        </ItemsListProvider>
    </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
});