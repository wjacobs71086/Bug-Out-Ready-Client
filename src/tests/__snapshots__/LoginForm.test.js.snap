// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<form
  className="LoginForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <input
    className="loginInput"
    name="user_name"
    placeholder="Username"
    required={true}
  />
  <input
    className="loginInput"
    name="password"
    placeholder="Password"
    required={true}
    type="password"
  />
  <button
    className="login"
    type="submit"
  >
    Login
  </button>
</form>
`;
